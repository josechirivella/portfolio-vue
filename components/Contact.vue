<template>
  <div id='contactForm' class='border border-gray-300 rounded-lg p-4 w-1/4'>
    <form class='card flex flex-col' @submit='submitForm'>
      <input id="name" v-model.lazy.trim="name" autocomplete="name" class="mb-4 p-2" placeholder="Your name"
             type="text">
      <input id="email" v-model.lazy.trim="email" autocomplete="email" class="mb-4 p-2" placeholder="Your email"
             type="email">
      <textarea id="message" v-model.lazy="message" class="p-2" placeholder="Enter your message">
      </textarea>
      <button>
        Submit
      </button>
    </form>
  </div>
</template>

<script>
export default {
  data() {
    return {
      name: '',
      email: '',
      message: ''
    };
  },
  methods: {
    submitForm(e) {
      e.preventDefault();
      const { name, email, message } = this;
      console.log({ name, email, message });
    }
  }
};
</script>
