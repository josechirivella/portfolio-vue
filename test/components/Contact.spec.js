import { beforeAll, describe, expect, test, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Contact from '~/components/Contact.vue';

describe.skip('Contact', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Contact, {
      data() {
        return {
          name: '',
          email: '',
          message: '',
        };
      },
    });
  });

  test('should render component', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  describe('Form elements', () => {
    test('form should exist', () => {
      expect(wrapper.find('form').exists()).toBe(true);
    });

    test('name input should exist', () => {
      expect(wrapper.find('#name').exists()).toBe(true);
    });

    test('email input should exist', () => {
      expect(wrapper.find('#email').exists()).toBe(true);
    });

    test('message textarea should exist', () => {
      expect(wrapper.find('#message').exists()).toBe(true);
    });

    test('submit button should exist', () => {
      expect(wrapper.find('button').exists()).toBe(true);
      expect(wrapper.find('button').text()).toBe('Submit');
    });
  });

  describe('Form submission', () => {
    test('submitForm should be called when form is submitted', async () => {
      const form = wrapper.find('form');
      const submitFormSpy = vi.spyOn(wrapper.vm, 'submitForm');

      await form.trigger('submit');

      expect(submitFormSpy).toHaveBeenCalled();
    });
  });
});
