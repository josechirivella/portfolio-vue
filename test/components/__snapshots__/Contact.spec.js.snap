// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact > should render component 1`] = `
<div
  class="border border-gray-300 rounded-lg p-4 w-1/4"
  id="contactForm"
>
  <form
    class="card flex flex-col"
  >
    <input
      autocomplete="name"
      class="mb-4 p-2"
      id="name"
      placeholder="Your name"
      type="text"
    />
    <input
      autocomplete="email"
      class="mb-4 p-2"
      id="email"
      placeholder="Your email"
      type="email"
    />
    <textarea
      class="p-2"
      id="message"
      placeholder="Enter your message"
    >

    </textarea>
    <button>
      Submit
    </button>
  </form>
</div>
`;
